custom_components/ai_thermostat/climate.py

```diff
@@ -275,7 +278,19 @@ def _async_startup(*_):
 					self._target_temp = float(old_state.attributes[ATTR_TEMPERATURE])
 			if not self._hvac_mode and old_state.state:
 				self._hvac_mode = old_state.state
-		
+			if old_state.attributes.get(ATTR_STATE_LAST_CHANGE) is not None:
```

Comment:
Just use `if not x:` instead of the lengthy `is not None`.

It will catch everything, like empty strings etc. too.